### PR TITLE
add unit tests

### PR DESCRIPTION
**Pull request by edaniszewski**
#72 

This PR adds a bunch of unit tests for:
- utils
- context commands

In addition, it makes some improvements to make testing easier and output more consistent, including:
- sorting context output
- reworking how the exiter utility behaves
- adding some test utilities to reduce duplicate code and make testing around Exit cases easier

There are still a bunch of tests which need to be written, primarily for the server commands and plugin commands. Will need to think through what the best way to do this is and how best to mock the server/plugin data.

### CODE REVIEW

**Comment by edaniszewski** [comment]
CI is failing because some of the Doc tests for output formatting do not give the expected output in CI, e.g.

```
--- FAIL: TestDoc (0.00s)

    doc_test.go:84: 

        	Error Trace:	doc_test.go:84

        	Error:      	Not equal: 

        	            	expected: "a string with \x1b[4munderscore\x1b[0m"

        	            	actual  : "a string with underscore"

        	            	

        	            	Diff:

        	            	--- Expected

        	            	+++ Actual

        	            	@@ -1 +1 @@

        	            	-a string with underscore

        	            	+a string with underscore

        	Test:       	TestDoc

        	Messages:   	string with console underscore marker

    doc_test.go:84: 

        	Error Trace:	doc_test.go:84

        	Error:      	Not equal: 

        	            	expected: "a string with \x1b[1mbold\x1b[0m"

        	            	actual  : "a string with bold"

        	            	

        	            	Diff:

        	            	--- Expected

        	            	+++ Actual

        	            	@@ -1 +1 @@

        	            	-a string with bold

        	            	+a string with bold

        	Test:       	TestDoc

        	Messages:   	string with console bold marker

    doc_test.go:84: 

        	Error Trace:	doc_test.go:84

        	Error:      	Not equal: 

        	            	expected: "a string with \x1b[0;36mcolor\x1b[0m"

        	            	actual  : "a string with color"

        	            	

        	            	Diff:

        	            	--- Expected

        	            	+++ Actual

        	            	@@ -1 +1 @@

        	            	-a string with color

        	            	+a string with color

        	Test:       	TestDoc

        	Messages:   	string with console color marker

```

Need to think about how to get these passing in CI..
